maitake/src/wait/cell.rs

```diff
@@ -189,6 +229,32 @@ impl fmt::Debug for WaitCell {
     }
 }
 
+// === impl Wait ===
+
+impl Future for Wait<'_> {
+    type Output = Result<(), Error>;
+
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        if !self.registered {
+            match self.cell.register_wait(cx.waker()) {
+                Ok(_) => {
+                    self.registered = true;
+                    Poll::Pending
+                }
+                Err(Error::Busy) => {
+                    // Cell was busy, all we can do is try again later
```

Comment:
This is where we could maybe have a configurable whether "busy" means "okay cool waited we're done here", or "keep trying". I think the as-is behavior is a reasonable position though.

Reply:
yeah, i think there could be a `wait_retrying` method (or something), and a bool on the future to configure the behavior. but, i think for now, let's not add stuff to the API unless we *need* to use it for something --- the current approach seems fine and we can always go back and make it configurable later.